### ISSUE TITLE
Support for Taikyoku shogi wanted

### ISSUE THREAD

**Issue by Karpeth** [u-fischer/chessboard#11]
While the package is great - there exists 1 chess game that cannot be represented with this package, Taikyoku shogi. Displaying the pieces is possible in XeLaTeX, as it is possible to define all 414 pieces needed using unicode, and the code still compiles.

However, as Taikyoku shogi uses a 36 by 36 board, it's impossible to render, as there is a check to convert the latin letter to a number.  

While I am not that great at the code behind the scenes, I can deduce that this code is what's causing the problem, and it should be easy to bypass or create a new key that bypasses this.

\newcommand*\board@conv@filechartonumber[2]{%
  \setcounter{#2}{\number`#1 }%
    \ifthenelse
     {\value{#2} = 90}%
     % char =Z
     {\addtocounter{#2}{-90}}%
     {\ifthenelse
       {\value{#2} >96 \AND \value{#2} <123}%
       %char a-z
       {\addtocounter{#2}{-96}}%
         {\PackageError{chessboard}%
           {Wrong field input}%
           {chars for files should be Z (for zero) or between a-z}}}}%
           
What I would propose as a solution is either a key (maxbigfield) where the file can be entered as a number, or a different workaround.

Feel free to mark this as completed as out of scope if it's too much.

**Comment by u-fischer** [comment]
please always provide a small but complete example. That makes it much easier to test an issue. (It doesn't need to make much sense, but it is a nuisance if I have to check first my own documentation to find out how to set the board size and trigger a specific error)

**Comment by Karpeth** [comment]
Working:
\documentclass[titlepage,a4paper,oneside]{report}
\usepackage{chessboard}
\begin{document}

\chessboard[maxfield=e36,addpieces=Kc3]
\end{document}
What I would want to work:
\documentclass[titlepage,a4paper,oneside]{report}
\usepackage{chessboard}
\begin{document}

\chessboard[maxfield={36}36,addpieces=Kc3]
\end{document}

**Comment by u-fischer** [comment]
Well you can try something like this. It will allow upto 52 fields (with the zero field). It is quite possible that somewhere in the code are more hidden assumptions on the a-z restriction ... (and redefining `\@alph` affects also the normal `\alph` but that is not trivial to avoid). 

~~~~
\documentclass[titlepage,a4paper,oneside]{report}
\usepackage{chessboard}
\makeatletter
\renewcommand*\board@conv@filechartonumber[2]{%
  \setcounter{#2}{\number`#1 }%
    \ifthenelse
     {\value{#2} = 90}%
     % char =Z
     {\addtocounter{#2}{-90}}%
     {\ifthenelse
       {\value{#2} >96 \AND \value{#2} <123}%
       %char a-z
       {\addtocounter{#2}{-96}}%
         {% A-Y
          \ifthenelse
           {\value{#2} >64 \AND \value{#2} <90}
           {\addtocounter{#2}{-38}}
           {\PackageError{chessboard}%
             {Wrong field input}%
             {chars for files should be Z (for zero) or between a-z and A-Y}}}}}%
\def\@zeroalph#1{%
  \ifcase#1Z\or a\or b\or c\or d\or e\or f\or g\or h\or i\or j\or
   k\or l\or m\or n\or o\or p\or q\or r\or s\or t\or u\or v\or w\or x\or
    y\or z \or A\or B\or C\or D\or E\or F\or G\or H\or I\or J\or K\or L\or M\or
    N\or O\or P\or Q\or R\or S\or T\or U\or V\or W\or X\or Y\else\@ctrerr\fi}%

\def\@alph#1{%
  \ifcase#1\or a\or b\or c\or d\or e\or f\or g\or h\or i\or j\or
   k\or l\or m\or n\or o\or p\or q\or r\or s\or t\or u\or v\or w\or x\or
    y\or z \or A\or B\or C\or D\or E\or F\or G\or H\or I\or J\or K\or L\or M\or
    N\or O\or P\or Q\or R\or S\or T\or U\or V\or W\or X\or Y\else\@ctrerr\fi}%
\makeatother             
\begin{document}

\chessboard[tinyboard,maxfield=J36,addpieces=KJ36]
\end{document}
~~~~

**Comment by Karpeth** [comment]
> Well you can try something like this. It will allow upto 52 fields (with the zero field). It is quite possible that somewhere in the code are more hidden assumptions on the a-z restriction ... (and redefining `\@alph` affects also the normal `\alph` but that is not trivial to avoid).
> 
> ```
> \documentclass[titlepage,a4paper,oneside]{report}
> \usepackage{chessboard}
> \makeatletter
> \renewcommand*\board@conv@filechartonumber[2]{%
>   \setcounter{#2}{\number`#1 }%
>     \ifthenelse
>      {\value{#2} = 90}%
>      % char =Z
>      {\addtocounter{#2}{-90}}%
>      {\ifthenelse
>        {\value{#2} >96 \AND \value{#2} <123}%
>        %char a-z
>        {\addtocounter{#2}{-96}}%
>          {% A-Y
>           \ifthenelse
>            {\value{#2} >64 \AND \value{#2} <90}
>            {\addtocounter{#2}{-38}}
>            {\PackageError{chessboard}%
>              {Wrong field input}%
>              {chars for files should be Z (for zero) or between a-z and A-Y}}}}}%
> \def\@zeroalph#1{%
>   \ifcase#1Z\or a\or b\or c\or d\or e\or f\or g\or h\or i\or j\or
>    k\or l\or m\or n\or o\or p\or q\or r\or s\or t\or u\or v\or w\or x\or
>     y\or z \or A\or B\or C\or D\or E\or F\or G\or H\or I\or J\or K\or L\or M\or
>     N\or O\or P\or Q\or R\or S\or T\or U\or V\or W\or X\or Y\else\@ctrerr\fi}%
> 
> \def\@alph#1{%
>   \ifcase#1\or a\or b\or c\or d\or e\or f\or g\or h\or i\or j\or
>    k\or l\or m\or n\or o\or p\or q\or r\or s\or t\or u\or v\or w\or x\or
>     y\or z \or A\or B\or C\or D\or E\or F\or G\or H\or I\or J\or K\or L\or M\or
>     N\or O\or P\or Q\or R\or S\or T\or U\or V\or W\or X\or Y\else\@ctrerr\fi}%
> \makeatother             
> \begin{document}
> 
> \chessboard[tinyboard,maxfield=J36,addpieces=KJ36]
> \end{document}
> ```

Thanks!

Just so I understand, a reason you would not merge this patch is because of \def\@alph?

**Comment by Karpeth** [comment]
Is there a reason for not using the https://ctan.org/pkg/alphalph package?

**Comment by u-fischer** [comment]
> Just so I understand, a reason you would not merge this patch is because of `\def\@alph`?

No. But I don't have the time now to track down all uses of `\alph` in the code and replace them by something else. And I don't have the time to test that, so I will wait if you report errors ...

> Is there a reason for not using the https://ctan.org/pkg/alphalph package?

The problem is not the representation but the parsing, I  don't want numbers consisting of two letters like `bc`.

**Comment by Karpeth** [comment]
Font shape `TU/lmss/m/up' undefined(Font) using `TU/lmss/m/n' instead

is the only warning I get just yet.
[testt.log](https://github.com/u-fischer/chessboard/files/14211314/testt.log)
[testt.pdf](https://github.com/u-fischer/chessboard/files/14211315/testt.pdf)

**Comment by Karpeth** [comment]
> Font shape `TU/lmss/m/up' undefined(Font) using `TU/lmss/m/n' instead
> 
> is the only warning I get just yet. [testt.log](https://github.com/u-fischer/chessboard/files/14211314/testt.log) [testt.pdf](https://github.com/u-fischer/chessboard/files/14211315/testt.pdf)

I much prefer the following:
\def\nyrank{\hfill\makebox[0pt][r]{\arabic{ranklabel}\hspace{1ex}}}

and using the hlabelformat=\nyrank. That does not cause overflow.

**Comment by Karpeth** [comment]
And now, a few months later, I have an error. The doubled size alph produces this when I try to make a 26-size table.

I'll try to follow up with an MWE.

> ! Package pgf Error: No shape named `z2' is known.
> 
> See the pgf package documentation for explanation.
> Type  H <return>  for immediate help.
>  ...                                              
>                                                   
> l.24 ...,Ξv2,Θv1,Ⲡw2,Ζw1,Rx2,Βx1,Py2,Ζy1,Τz2,Ⲛz1}]
>                                                    Promoverad
> This error message was generated by an \errmessage
> command, so I can't give any explicit help.
> Pretend that you're Hercule Poirot: Examine all clues,
> and deduce the truth by order and method.
